### PR TITLE
all work component with changes

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3986
https://tools.hmcts.net/jira/browse/EUI-3979
https://tools.hmcts.net/jira/browse/EUI-3933
https://tools.hmcts.net/jira/browse/EUI-3935
https://tools.hmcts.net/jira/browse/EUI-3644
https://tools.hmcts.net/jira/browse/EUI-3934
https://tools.hmcts.net/jira/browse/EUI-4228

### Change description ###

WA All Work Functionality

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by connorpgpmcelroy — APPROVED** [review]
Approved, on condition that the PR environment builds without issue, because I have understanding of the code and having looked at it thoroughly, there is nothing that looks out of place or incorrect to me. Any changes that I can see that will eventually need to be made will be addressed in a future PR made by me.

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
+import { CdkTableModule } from '@angular/cdk/table';
+import { Component, ViewChild } from '@angular/core';
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Router} from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AlertService, LoadingService, PaginationModule } from '@hmcts/ccd-case-ui-toolkit';
+import { ExuiCommonLibModule, FeatureToggleService } from '@hmcts/rpx-xui-common-lib';
+import { of } from 'rxjs';
+import { SessionStorageService } from 'src/app/services';
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/all-work-task/all-work-task.component.spec.ts:9]: src

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts [hunk 2]
```diff
@@ -1,109 +1,58 @@
-import { Component, OnInit } from '@angular/core';
-import { Router } from '@angular/router';
-import { Observable } from 'rxjs';
+import { Component } from '@angular/core';
 import { UserInfo } from 'src/app/models/user-details.model';
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:2]: src

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts [hunk 3]
```diff
@@ -1,109 +1,58 @@
...
-import { SessionStorageService } from 'src/app/services';
-import { CONFIG_CONSTANTS } from 'src/work-allocation-2/components/constants/config.constants';
-import { LIST_CONSTANTS } from 'src/work-allocation-2/components/constants/list.constants';
-import { TaskService, TaskSort } from 'src/work-allocation-2/enums';
-import { PaginationParameter, SearchTaskRequest, SortParameter } from 'src/work-allocation-2/models/dtos';
-import { WorkAllocationTaskService } from 'src/work-allocation-2/services';
-import { handleFatalErrors, WILDCARD_SERVICE_DOWN } from 'src/work-allocation-2/utils';
-import { InvokedTaskAction, Task, TaskFieldConfig, TaskServiceConfig, TaskSortField } from '../../models/tasks';
+import { ConfigConstants, ListConstants, PageConstants, SortConstants } from 'src/work-allocation-2/components/constants';
+import { SearchTaskRequest } from 'src/work-allocation-2/models/dtos';
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:4]: src

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts [hunk 4]
```diff
@@ -1,109 +1,58 @@
...
 import { UserInfo } from 'src/app/models/user-details.model';
-import { SessionStorageService } from 'src/app/services';
-import { CONFIG_CONSTANTS } from 'src/work-allocation-2/components/constants/config.constants';
-import { LIST_CONSTANTS } from 'src/work-allocation-2/components/constants/list.constants';
-import { TaskService, TaskSort } from 'src/work-allocation-2/enums';
-import { PaginationParameter, SearchTaskRequest, SortParameter } from 'src/work-allocation-2/models/dtos';
-import { WorkAllocationTaskService } from 'src/work-allocation-2/services';
-import { handleFatalErrors, WILDCARD_SERVICE_DOWN } from 'src/work-allocation-2/utils';
-import { InvokedTaskAction, Task, TaskFieldConfig, TaskServiceConfig, TaskSortField } from '../../models/tasks';
+import { ConfigConstants, ListConstants, PageConstants, SortConstants } from 'src/work-allocation-2/components/constants';
```
Comment (rayliangatos) [inline src/work-allocation-2/containers/all-work-task/all-work-task.component.ts:3]: src

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.spec.ts (hunk 1 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.spec.ts]: Changed @rayliangatos

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts (hunk 2 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.ts]: Changed @rayliangatos

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts (hunk 3 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.ts]: Changed @rayliangatos

File: src/work-allocation-2/containers/all-work-task/all-work-task.component.ts (hunk 4 above)
Comment (udaydenduluri33) [reply src/work-allocation-2/containers/all-work-task/all-work-task.component.ts]: Changed @rayliangatos

**Review by rayliangatos — APPROVED** [review]

**Pull request merged**